package-res/ccc/core/base/axis/color-axis.js

```diff
@@ -60,11 +71,81 @@ def('pvc.visual.ColorAxis', pvc_Axis.extend({
             return this.base(scale);
         },
 
+        // CDF603 
+        // set PreserveMap default to true
+        // must be called before the first read of PreserveMap? 
+        setPreserveColorMap: function() {
+            this.option.defaults({'PreserveMap': true});
+        },
+
+
+        // CDF603
+        // returns the stored Map if it is supposed to be 
+        // preserved and if it exists
+        _haveMap: function() {
+              if(this.option('PreserveMap') && this.Map) 
+                return this.Map;  
+        },
+
+ 
+        //CDF603
+        // given a colorMap it substitutes it if it exists
+        // a previously stored one
+        effectiveMap: function(colorMap , baseScheme) {
+            var prevMap = this._haveMap(),
+                newMap;
+
+            if(prevMap) newMap = prevMap;
+            else newMap = colorMap ;
+
+            return newMap;
+        },
+
+        //CDF603
+        // given a scheme, computes the corresponding colorMap
+        // scheme is a function that for a given element of the 
```

Comment:
Actually, in CCC, a color scheme is a color scale factory - it creates a new, reset, color scale function, each time it is called. What is describe as a scheme is just a color scale :-) Please change the function and argument names accordingly.